app/src/main/java/com/example/calculator_final/FragmentKeyboardFirst.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.calculator_final
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import kotlinx.android.synthetic.main.fragment_keyboard_1.*
+
+class FragmentKeyboardFirst : Fragment() {
+
+    enum class ButtonType (){
+        NUMBER,
+        OPERATOR,
+        DELETE,
+        CLEAR,
+        CALCULATING,
+        CHANGEKEYBOARD
+    }
+
+    var communicator: Communicator? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        communicator = context as Communicator
+        return inflater.inflate(R.layout.fragment_keyboard_1, container, false)
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        // Number
+        buttonNumZero.setOnClickListener { communicator?.respondData(ButtonType.NUMBER, "0") }
```

Comment:
```suggestion
        buttonNumZero.setOnClickListener { communicator?.respondData(ButtonType.NUMBER, it.text) }
```
Mỗi button đều có text của nó rồi thì em nên tận dụng